core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/ArchiveManipulator.java

```diff
@@ -166,6 +166,13 @@ public void extractFromStream(InputStream inputStream, String extractDir) throws
                 String entryName = entry.getName();
                 File f = new File(extractDir + File.separator + entryName);
 
+                String canonicaltargetDirectoryPath = new File(extractDir).getCanonicalPath();
+                String canonicalFilePath = f.getCanonicalPath();
+                if (!canonicalFilePath.startsWith(canonicaltargetDirectoryPath)) {
+                    throw new IOException("Attempt to upload invalid zip archive with file at " + f + ". File path is " +
+                            "outside target directory");
+                }
+
```

Comment:
it seems we are catching the io exception within the catch block. do we need to throw io exception or just a exception here

Reply:
If this is changed to an exception instead, we have to modify the method signature and handle the exception at all places this method is called. So for the moment, we shall keep this as it is.